src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -5322,7 +5387,40 @@ BOOL HandleHardwareException(PAL_SEHException* ex)
             fef.InitAndLink(ex->GetContextRecord());
         }
 
-        DispatchManagedException(*ex, true /* isHardwareException */);
+        if (g_isNewExceptionHandlingEnabled)
+        {
+            REGDISPLAY rd;
+            Thread *pThread = GetThread();
+
+            CONTEXT ctx = {0};
+            ctx.ContextFlags = CONTEXT_CONTROL | CONTEXT_INTEGER;
+            ExInfo exInfo(pThread, &ctx /*ex->GetContextRecord()*/, &rd, ExKind::HardwareFault);
+
+            DWORD exceptionCode = ex->GetExceptionRecord()->ExceptionCode;
+            if (exceptionCode == STATUS_ACCESS_VIOLATION)
+            {
+                #define NULL_AREA_SIZE   (64 * 1024)
```

Comment:
This is Windows-specific definition that is not correct for Unix. Move the definition from excep.cpp to a shared place instead.